src/blocks/networkmanager.rs

```diff
@@ -28,125 +32,371 @@ enum NetworkState {
+    BT = 5,
+    OLPCMesh = 6,
+    WiMAX = 7,
+    Modem = 8,
+    InfiniBand = 9,
+    Bond = 10,
+    VLAN = 11,
+    ADLS = 12,
+    Bridge = 13,
+    Generic = 14,
+    Team = 15,
+    TUN = 16,
+    IPTunnel = 17,
+    MACVLAN = 18,
+    VXLAN = 19,
+    VETH = 20,
+    MACsec = 21,
+    Dummy = 22,
+    PPP = 23,
+    OVSInterface = 24,
+    OVSPort = 25,
+    OVSBridge = 26,
+    WPAN = 27,
+    IPv6LoWPAN = 28,
+    Wireguard = 29,
 }
 
-impl From<String> for ConnectionType {
-    fn from(name: String) -> Self {
-        match name.as_ref() {
-            // https://developer.gnome.org/NetworkManager/unstable/settings-connection.html
-            "802-3-ethernet" => ConnectionType::Ethernet,
-            "802-11-wireless" => ConnectionType::Wireless,
-            _  => ConnectionType::Other,
+impl From<u32> for DeviceType {
+    fn from(id: u32) -> Self {
+        match id {
+            // https://developer.gnome.org/NetworkManager/stable/nm-dbus-types.html#NMDeviceType
+            1 => DeviceType::Ethernet,
+            2 => DeviceType::Wifi,
+            3 => DeviceType::Unused1,
+            4 => DeviceType::Unused2,
+            5 => DeviceType::BT,
+            6 => DeviceType::OLPCMesh,
+            7 => DeviceType::WiMAX,
+            8 => DeviceType::Modem,
+            9 => DeviceType::InfiniBand,
+            10 => DeviceType::Bond,
+            11 => DeviceType::VLAN,
+            12 => DeviceType::ADLS,
+            13 => DeviceType::Bridge,
+            14 => DeviceType::Generic,
+            15 => DeviceType::Team,
+            16 => DeviceType::TUN,
+            17 => DeviceType::IPTunnel,
+            18 => DeviceType::MACVLAN,
+            19 => DeviceType::VXLAN,
+            20 => DeviceType::VETH,
+            21 => DeviceType::MACsec,
+            22 => DeviceType::Dummy,
+            23 => DeviceType::PPP,
+            24 => DeviceType::OVSInterface,
+            25 => DeviceType::OVSPort,
+            26 => DeviceType::OVSBridge,
+            27 => DeviceType::WPAN,
+            28 => DeviceType::IPv6LoWPAN,
+            29 => DeviceType::Wireguard,
+            _ => DeviceType::Unknown,
         }
     }
 }
 
-impl fmt::Display for ConnectionType {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+impl DeviceType {
+    fn to_icon_name(&self) -> Option<String> {
         match self {
-            ConnectionType::Ethernet => write!(f, "net_wired"),
-            ConnectionType::Wireless => write!(f, "net_wireless"),
-            ConnectionType::Other => write!(f, "net_other"),
+            DeviceType::Ethernet => Some("net_wired".to_string()),
+            DeviceType::Wifi => Some("net_wireless".to_string()),
+            DeviceType::Modem => Some("net_modem".to_string()),
+            DeviceType::Bridge => Some("net_bridge".to_string()),
+            DeviceType::TUN => Some("net_bridge".to_string()),
+            DeviceType::Wireguard => Some("net_vpn".to_string()),
+            _ => None,
```

Comment:
This list of device types looks more reasonable to me. Most of the others should be removed.

Reply:
Yup